### PR TITLE
[Merged by Bors] - feat(MeasureTheory): product of bases and Basis.addHaar commute

### PR DESCRIPTION
**Pull request by mcdoll**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean [hunk 1]
```diff
@@ -247,6 +277,12 @@ theorem Basis.addHaar_self (b : Basis ι ℝ E) : b.addHaar (_root_.parallelepip
   rw [Basis.addHaar]; exact addHaarMeasure_self
 #align basis.add_haar_self Basis.addHaar_self
 
+variable [MeasurableSpace F] [BorelSpace F] [SecondCountableTopology E] [SecondCountableTopology F]
+
+theorem Basis.prod_addHaar (v : Basis ι ℝ E) (w : Basis ι' ℝ F) :
+    (v.prod w).addHaar = v.addHaar.prod w.addHaar := by
+  simp [(v.prod w).addHaar_eq_iff, Basis.prod_parallelepiped, Basis.addHaar_self]
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean:284]:
```suggestion
variable [MeasurableSpace F] [BorelSpace F] [SecondCountableTopology E]

theorem Basis.prod_addHaar (v : Basis ι ℝ E) (w : Basis ι' ℝ F) :
    (v.prod w).addHaar = v.addHaar.prod w.addHaar := by
  have : FiniteDimensional ℝ F := FiniteDimensional.of_fintype_basis w
  simp [(v.prod w).addHaar_eq_iff, Basis.prod_parallelepiped, Basis.addHaar_self]
```
Could you also use the same trick in the instance `SigmaFinite b.addHaar` a few lines above?

**Comment by sgouezel** [comment]
bors d+

File: Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean (hunk 1 above)
Comment (mcdoll) [reply Mathlib/MeasureTheory/Measure/Haar/OfBasis.lean]: I made it more symmetric.

**Comment by mcdoll** [comment]
bors merge

**Pull request closed without merging**